tomviz/DataPropertiesPanel.cxx

```diff
@@ -131,6 +131,43 @@ QString getDataDimensionsString(vtkSMSourceProxy* proxy)
   return extentString;
 }
 
+template <typename T>
+QString getSizeNearestThousand(T num)
+{
+  char format = 'f';
+  int prec = 1;
+
+  QString ret;
+  if (num < 1e3)
+    ret = QString::number(num) + " B";
+  else if (num < 1e6)
+    ret = QString::number(num / 1e3, format, prec) + " K";
+  else if (num < 1e9)
+    ret = QString::number(num / 1e6, format, prec) + " M";
+  else if (num < 1e12)
+    ret = QString::number(num / 1e9, format, prec) + " G";
+  else
```

Comment:
We could add T, and why not just leave the final statement as / 1e12 etc and then let it be bigger if it somehow loads?

Reply:
Done.